returns/primitives/reawaitable.py

```diff
@@ -2,6 +2,16 @@
 from functools import wraps
 from typing import NewType, ParamSpec, TypeVar, cast, final
 
+# Try to use anyio.Lock, fall back to asyncio.Lock
+try:
+    import anyio  # noqa: WPS433
+except ImportError:  # pragma: no cover
+    import asyncio  # noqa: WPS433
+
+    Lock = asyncio.Lock
+else:
+    Lock = anyio.Lock
```

Comment:
You can type it as a simple interface that only supports `async with`, probably like `contextlib.AbstractAsyncContextManager`